.github/workflows/semantic-commit.yml

```diff
@@ -0,0 +1,41 @@
+name: Semantic Commit Check
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened, edited]
+
+jobs:
+  semantic-commit:
+    runs-on: ubuntu-latest
+    name: Check Semantic Commits
+    steps:
+      - name: Checkout repository
```

Comment:
Checkout of the repository is optional step.
If your action does not need repository (uses GH API you can skip it)